src/skin.cpp

```diff
@@ -431,7 +416,7 @@ void SkinInfillAreaComputation::generateInfill(SliceLayerPart& part, const Polyg
     const size_t wall_line_count = mesh.settings.get<size_t>("wall_line_count");
     const coord_t infill_line_distance = mesh.settings.get<coord_t>("infill_line_distance");
 
-    coord_t offset_from_inner_wall = -infill_skin_overlap;
+    coord_t offset_from_inner_wall = 0;
```

Comment:
Good practice imo to make a lambda for such a calculation. This way the `offset_from_inner_wall` can be made const, and const is good. Newer compilers will be able to optimize the hell out of that. It will also be clear to readers of this code in the future what the actual purpose is of this piece of code.
 
```c++
auto get_offset = [](const Settings& settings)
{
    const auto wall_line_count = settings.get<size_t>("wall_line_count");
    const auto wall_line_count = settings.get<size_t>("wall_line_count");
    if (wall_line_count > 0)
    { // calculate offset_from_inner_wall
        coord_t extra_perimeter_offset = 0; // to align concentric polygons across layers
        const auto fill_pattern = settings.get<EFillMethod>("infill_pattern");
        if (fill_pattern == EFillMethod::CONCENTRIC
            && infill_line_distance > settings.get<coord_t>("infill_line_width") * 2)
        {
            if (settings.get<bool>("alternate_extra_perimeter")
                && layer_nr % 2 == 0)
            { // compensate shifts otherwise caused by alternating an extra perimeter
                extra_perimeter_offset = -innermost_wall_line_width;
            }
            if (layer_nr == 0)
            { // compensate for shift caused by walls being expanded by the initial line width multiplier
                const auto normal_wall_line_width_0 = settings.get<coord_t>("wall_line_width_0");
                const auto normal_wall_line_width_x = settings.get<coord_t>("wall_line_width_x");
                const coord_t normal_walls_width = normal_wall_line_width_0 + (wall_line_count - 1) * normal_wall_line_width_x;
                const coord_t walls_width = normal_walls_width * settings.get<Ratio>("initial_layer_line_width_factor");
                extra_perimeter_offset += walls_width - normal_walls_width;
                while (extra_perimeter_offset > 0)
                {
                    extra_perimeter_offset -= infill_line_distance;
                }
            }
        }
        offset_from_inner_wall += extra_perimeter_offset - innermost_wall_line_width / 2;
    }
};

const coord_t offset_from_inner_wall = get_offset(mesh.settings);
...
```